### PR TITLE
dev flake

### PR DESCRIPTION
**Pull request by zowoq**
This allows us to have private dev inputs that we can use in the main flake and move the github actions checks to buildbot.

We may be able to upstream the vendored files to the flake-parts repo but I don't want bother with that for now.

~~Depends on https://github.com/nix-community/srvos/pull/355.~~

### CODE REVIEW

**Review by zimbatm — CHANGES_REQUESTED** [review]
What problem does it solve? Please add more details to your commit messages so it's possible to follow the reasoning.

**Comment by zowoq** [comment]
It allows us to have private dev inputs that we can use in the main flake, can then move the github actions checks to buildbot. We discussed this briefly on matrix a few weeks ago.

**Comment by zowoq** [comment]
I plan to merge this before the next flake update.

File: .github/workflows/update-flake-lock.yml [hunk 1]
```diff
@@ -12,9 +12,25 @@ jobs:
...
+          pushd dev/private
+          nix flake update --commit-lock-file
+          popd
+          nix run .#update-dev-private-narHash
+          git commit --amend --no-edit dev/private.narHash
+      - uses: peter-evans/create-pull-request@v5
+        # disable for now
+        #with:
+        #  labels: |
+        #    merge-queue
```
Comment (zowoq) [inline .github/workflows/update-flake-lock.yml:36]: I'll merge these manually for now, will re-enable this in a week or two.

**Pull request merged**